lib/content-types/merge.js

```diff
@@ -155,6 +168,34 @@ const squish = (types) => {
         return plugin;
       });
 
+      if (mergedType.identifier) {
```

Comment:
It does happen in `load`. Or rather, the `check` function is called in `load`, which has a [whole section on checking `identifier`](https://github.com/punchcard-cms/content-types/pull/72/files#diff-23919d27ac0348e07fd7187ffdfdf6c6R110).

This could not happen exclusively in `load` as `load` does not know the details about the individual input plugins.